plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/procedure/FlushHiveMetastoreCacheProcedure.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.procedure;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.plugin.hive.metastore.cache.CachingHiveMetastore;
+import io.trino.spi.procedure.Procedure;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import java.lang.invoke.MethodHandle;
+
+import static io.trino.spi.block.MethodHandleUtil.methodHandle;
+import static java.util.Objects.requireNonNull;
+
+public class FlushHiveMetastoreCacheProcedure
+        implements Provider<Procedure>
+{
+    private static final MethodHandle FLUSH_HIVE_METASTORE_CACHE = methodHandle(
+            FlushHiveMetastoreCacheProcedure.class,
+            "flushHiveMetastoreCache");
+
+    private final HiveMetastore hiveMetastore;
+
+    @Inject
+    public FlushHiveMetastoreCacheProcedure(HiveMetastore hiveMetastore)
+    {
+        this.hiveMetastore = requireNonNull(hiveMetastore, "hiveMetastore is null");
+    }
+
+    @Override
+    public Procedure get()
+    {
+        return new Procedure(
+                "system",
+                "flush_hive_metastore_cache",
+                ImmutableList.of(),
+                FLUSH_HIVE_METASTORE_CACHE.bindTo(this));
+    }
+
+    public void flushHiveMetastoreCache()
+    {
+        if (hiveMetastore instanceof CachingHiveMetastore) {
```

Comment:
- I do think `CachingHiveMetastore` is always bound in guice, so inject it directly.
- `instanceof` is error prone. For example, there might be additional wrapper around `CachingHiveMetastore` (like recording metastore, or whatever), and then this wouldn't work. Worse, it could work or not, depending on configuration.

Reply:
Will try but there is no direct binding to CachingHiveMetastore will add it then.

Reply:
I think we should add an optional binder to `CachingHiveMetastore` which is bound only if the metastore is "caching" (even if wrapped in something).
Alternatively we can add separate `HiveMetastoreCacheFlusher` interface and add optional binder for that one. But it may be an overkill.

Reply:
I've adde `Optional<CachingHiveMetastore>` binding